### PR TITLE
#554 Add an ideal switch model

### PR DESCRIPTION
**Pull request by adriguir**

### CODE REVIEW

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Switches/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+# See AUTHORS.txt
+# All rights reserved.
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, you can obtain one at http://mozilla.org/MPL/2.0/.
+# SPDX-License-Identifier: MPL-2.0
+#
+# This file is part of Dynawo, an hybrid C++/Modelica open source time domain simulation tool for power systems.
+
+# Transformers models
```
Comment (MarianneSaugierRTE) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Switches/CMakeLists.txt]: comment

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Switches/IdealSwitch.mo [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+*
+* This file is part of Dynawo, an hybrid C++/Modelica open source time domain simulation tool for power systems.
+*/
+
+model IdealSwitch "Ideal switch"
+  /*
+    When the switch is closed, the voltage on both terminals are equal and the current is going through the switch.
+    When the switch is open, the current going through the switch is equal to zero.
+
+    Equivalent circuit and conventions:
```
Comment (MarianneSaugierRTE) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Switches/IdealSwitch.mo]: good idea this graphical view of the conventions!

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Switches/IdealSwitch.mo [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+               I1                  I2
+    (terminal1) -->-------/ -------<-- (terminal2)
+
+  */
+
+  import Dynawo.Connectors;
+
+  Connectors.ACPower terminal1 "Switch side 1";
+  Connectors.ACPower terminal2 "Switch side 2";
+  Connectors.BPin closed (value(start = true)) "True when the switch is closed, false otherwise";
```
Comment (MarianneSaugierRTE) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Switches/IdealSwitch.mo]: Should be use the state enum (open, closed, closed1, closed2) and have something in Controls>Basics>SwitchOff as for lines and transformers?

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Switches/CMakeLists.txt (hunk 1 above)
Comment (adriguir) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Switches/CMakeLists.txt]: Corrected

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Switches/IdealSwitch.mo (hunk 3 above)
Comment (adriguir) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Switches/IdealSwitch.mo]: Done

**Pull request merged**